Update MacOS build image

Available build images:
https://circleci.com/docs/using-macos#supported-xcode-versions

----

PR #1034 recently updated the MacOS build image to 12.5.1. There is a stable 13.4.1 currently available. There is also a 14.0.0 Beta 5 release.

----

We can't currently upgrade to 13.4.1, since that causes the build to fail.

Example of failure:
https://app.circleci.com/pipelines/github/lairworks/nas2d-core/1986/workflows/f4e7c376-605c-4b7b-beb4-f10ed6a235f0/jobs/8179

> In file included from NAS2D/FpsCounter.cpp:13:
> In file included from /usr/local/include/SDL2/SDL.h:41:
> /usr/local/include/SDL2/SDL_events.h:660:65: error: use of old-style cast [-Werror,-Wold-style-cast]
> ```cpp
> SDL_COMPILE_TIME_ASSERT(SDL_Event, sizeof(SDL_Event) == sizeof(((SDL_Event *)NULL)->padding));
>                                                                 ^            ~~~~
> /usr/local/include/SDL2/SDL_stdinc.h:373:57: note: expanded from macro 'SDL_COMPILE_TIME_ASSERT'
> #define SDL_COMPILE_TIME_ASSERT(name, x)  static_assert(x, #x)
>                                                         ^
> ```

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -18,7 +18,7 @@ commands:
 jobs:
   build-macos:
     macos:
-      xcode: "12.5.1"
+      xcode: "13.4.1"
     environment:
       - HOMEBREW_NO_AUTO_UPDATE: 1
       - WARN_EXTRA: "-isystem /usr/local/include"

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -21,7 +21,7 @@ jobs:
       xcode: "12.5.1"
     environment:
       - HOMEBREW_NO_AUTO_UPDATE: 1
-      - WARN_EXTRA: "-Wno-double-promotion"
+      - WARN_EXTRA: "-isystem /usr/local/include"
     steps:
       - checkout
       - brew-install

